server/index.js

```diff
@@ -176,6 +176,43 @@ app.post('/api/cart/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/orders', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!(cartId)) {
+    return res.status(400).json({
+      error: "A 'cartId' was not found in your session, please contact Site Admin for assistance."
+    });
+  }
+  const name = req.body.name;
+  const creditCard = req.body.creditCard;
+  const shippingAddress = req.body.shippingAddress;
+  if (!(name && creditCard && shippingAddress)) {
+    return res.status(400).json({
+      error: 'Customer information is missing, please make sure all customer details have been entered.'
+    });
+  }
+  const sql = `
+  insert into "orders" ("cartId", "name", "creditCard", "shippingAddress")
+                values ($1, $2, $3, $4)
+             returning *
+  `;
+  const values = [cartId, name, creditCard, shippingAddress];
+  db.query(sql, values)
+    .then(result => {
+      if (result.rows[0]) {
+        delete req.session.cartId;
+      }
```

Comment:
If the insert succeeds, then the first row returned will always be truthy. Otherwise, the `.then()` would be skipped and `.catch()` would fire instead.

Reply:
ahh that makes so much sense, thanks Tim.  I removed the conditional statement, and i tested it with an invalid query instead, and .catch() was then called.